R/load_package.r

```diff
@@ -1,15 +1,89 @@
 #' Load a package and install it if necessary
 #' 
 #' @name load_package
-#' @param name Name of package
+#' @param name character. Name of package.
+#' @param verbose logical. Whether or not to announce each installation.
 #' @examples
 #' \dontrun{
-#' load_package('glmnet')
+#' load_package("glmnet")
+#' load_package("robertzk/Ramd")
+#' load_package("robertzk/Ramd@@v0.3")   # Can load from versions
+#' load_package("robertzk/Ramd@@0.3")
+#' load_package("robertzk/Ramd@@fbe1aa0e36df289b27881d077635352e6debdbc1")  # Can load from refs
+#' load_package(list("FeiYeYe/xgboost", subdir = "R-package"))  # Can load from subdirectories
 #' }
-load_package <- function(name) {
-  if (!require(name, character.only = TRUE)) {
-    install.packages(name, dependencies = TRUE)
-    if (!require(name, character.only = TRUE))
-      stop(paste('Package', name, 'not found'))
+load_package <- function(name, verbose = FALSE) {
+  if (is.list(name)) {
+    metadata <- name[-1]  # For tracking things like subdir
+    name <- name[[1]]
+  } else {
+    metadata <- NULL
   }
+  if (package_is_installed(name)) {
+    if (isTRUE(verbose)) { message(name, " already installed.") }
+    return(TRUE)
+  }
+  if (is_version_mismatch(name)) {
+    if (isTRUE(verbose)) {
+      message("Removing prior insallation of ", name_from_github_name(name))
+    }
+    utils::remove.packages(name)
+  }
+  if (is_github_package(name)) {
```

Comment:
I would make this whole section from lines 32 to 45 a separate method, like `install_package`.